src/common/libjob/job.c

```diff
@@ -658,6 +659,218 @@ int flux_job_id_encode (flux_jobid_t id,
     return fluid_encode (buf, bufsz, id, t);
 }
 
+static flux_job_result_t job_result_calc (json_t *res)
+{
+    double t_run = -1.;
+    int status = -1;
+    bool exception = false;
+    const char *exception_type = NULL;
+    json_error_t error;
+
+    if (json_unpack_ex (res, &error, 0,
+                        "{s?f s:b s?i s?s}",
+                        "t_run", &t_run,
+                        "exception_occurred", &exception,
+                        "waitstatus", &status,
+                        "exception_type", &exception_type) < 0)
+        return FLUX_JOB_RESULT_FAILED;
+
+    if (t_run > 0. && status == 0)
+        return FLUX_JOB_RESULT_COMPLETED;
+    if (exception) {
+        if (exception_type != NULL) {
+            if (strcmp (exception_type, "cancel") == 0)
+                return FLUX_JOB_RESULT_CANCELED;
+            if (strcmp (exception_type, "timeout") == 0)
+                return FLUX_JOB_RESULT_TIMEOUT;
+        }
+    }
+    return FLUX_JOB_RESULT_FAILED;
+}
+
+static void result_eventlog_error_cb (flux_future_t *f, void *arg)
+{
+    json_t *res = arg;
+    char *s = NULL;
+
+    if (flux_future_get (f, NULL) < 0 && errno != ENODATA) {
+        flux_future_continue_error (f, errno, NULL);
+        flux_future_destroy (f);
+        return;
+    }
+    flux_job_result_t result = job_result_calc (res);
+    if (json_object_set_new (res,
+                             "result",
+                             json_integer (result)) < 0)
```

Comment:
check for NULL return of `json_integer()`

is it ok to just return here if there is an error?

Reply:
Good catch. The future should be fulfilled with an error on failure here.